packages/@aws-cdk/aws-iot-actions/lib/location-action.ts

```diff
@@ -0,0 +1,160 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CfnTopicRule } from '@aws-cdk/aws-iot';
+import { CfnTracker } from '@aws-cdk/aws-location';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Location Timestamp Unit Enum
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export enum LocationTimestampUnit {
+  /**
+   * SECONDS
+   */
+  SECONDS='SECONDS',
+
+  /**
+   * MILLISECONDS
+   */
+  MILLISECONDS='MILLISECONDS',
+
+  /**
+   * MICROSECONDS
+   */
+  MICROSECONDS='MICROSECONDS',
+
+  /**
+   * NANOSECONDS
+   */
+  NANOSECONDS='NANOSECONDS',
+}
+
+/**
+ * Describes how to interpret an application-defined timestamp value from an MQTT message payload and the precision of that value
+ *
+ * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+ */
+export interface LocationTimestamp {
+  /**
+     * The precision of the timestamp value that results from the expression described in `value`.
+     * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationTimestamp.html
+     *
+     * @default MILLISECONDS
+     */
+  readonly unit?: LocationTimestampUnit
+
+  /**
+     * An expression that returns a long epoch time value
+     * @see https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html
+     *
+     * @default '${timestamp()}'
+     */
+  readonly value?: string
+}
+
+/**
+ * Configuration properties of an action for Location.
+ */
+export interface LocationActionProps extends CommonActionProps {
+  /**
+   * The deviceId providing the location data.
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   *
+   * @default None
+   */
+  readonly deviceId: string;
+
+  /**
+   * A string that evaluates to a double value that represents the latitude of the device's location
+   * @see https://docs.aws.amazon.com/iot/latest/apireference/API_LocationAction.html
+   *
+   * @default None
+   */
+  readonly latitude: string;
```

Comment:
This gets converted to a string but it looks like they expect a double in the form of a string. Let's have users input a number and then we'll parse it for them into the string.

Reply:
From the documentation it says that we provide it as a string and it will be evaluated to a double value. More importantly, we need it to be a string as we can use the [IoT substitution template](https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html). 

For example, if we receive a message such as the following on topic `dp/123456/data`:
```
{
   "latitude": 40.7484405,
   "longitude": -73.9856644,
}
```

Our SQL statement is just `SELECT ${topic(2} as device_id, * FROM dp/+/data`, then our IoT Location action will need to look like this:
```
{
      Location: {
        DeviceId: '${device_id}',
        Latitude: '${latitude}',
        Longitude: '${longitude}',
        RoleArn: 'theRoleArnToUseHere',
        Timestamp:  {
            Value: '${timestamp()}',
            Unit: LocationTimestampUnit.MILLISECONDS,
        },
        TrackerName: 'someTracker',
      },
}
```
in the console it would look as follows
![Bildschirm­foto 2023-02-10 um 09 35 44](https://user-images.githubusercontent.com/4415930/218042956-13f84dbc-a902-485e-bcbf-5e50779ad79a.png)